src/models/room.ts

```diff
@@ -1880,6 +1888,41 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         this.threadsReady = true;
     }
 
+    public processPollEvents(events: MatrixEvent[], matrixClient: MatrixClient): void {
+        const processPollStartEvent = (event: MatrixEvent): void => {
+            if (!M_POLL_START.matches(event.getType())) return;
+            const poll = new Poll(event, matrixClient);
+            this.polls.set(event.getId()!, poll);
+            this.emit(PollEvent.New, poll);
+        };
+
+        const processPollRelationEvent = (event: MatrixEvent): void => {
+            const relationEventId = event.getRelation()?.event_id;
+            if (relationEventId && this.polls.has(relationEventId)) {
+                const poll = this.polls.get(relationEventId);
+                poll?.onNewRelation(event);
+            }
+        };
+
+        const processPollEvent = (event: MatrixEvent): void => {
+            processPollStartEvent(event);
+            processPollRelationEvent(event);
+        };
+
+        events.forEach((event: MatrixEvent) => {
+            matrixClient.decryptEventIfNeeded(event);
+
+            if (event.isBeingDecrypted() || event.isDecryptionFailure()) {
+                // add an event listener for once the event is decrypted.
+                event.once(MatrixEventEvent.Decrypted, async () => {
+                    processPollEvent(event);
+                });
+            } else {
+                processPollEvent(event);
+            }
+        });
+    }
```

Comment:
```suggestion
	for (const event of events) {
            try {
                await matrixClient.decryptEventIfNeeded(event);
                processPollEvent(event);
            } catch {}
        }
```

`decryptEventIfNeeded` returns a resolved promise for unencrypted or already decrypted events. The above might read a bit better. But that surely shouldn't be blocking this pull request